test/test_models_recognition.py

```diff
@@ -52,6 +52,23 @@ def test_recognition_models(arch_name, input_shape, output_size):
     assert out.numpy().shape == (batch_size, *output_size)
 
 
+def test_sar_training():
+    batch_size = 4
+    input_shape = (64, 256, 3)
+    output_size = (41, 119)
+    reco_model = recognition.sar_vgg16_bn(input_shape=input_shape)
+    input_tensor = tf.random.uniform(shape=[batch_size, *input_shape], minval=0, maxval=1)
+    indices = [[0, 0], [0, 1], [1, 0], [1, 1], [1, 2], [2, 0], [3, 0], [3, 1], [3, 2], [3, 3], [3, 4]]
+    values = tf.random.uniform(shape=[11], minval=0, maxval=118, dtype=tf.dtypes.int64)
+    input_labels = tf.sparse.reorder(
+        tf.sparse.SparseTensor(indices=indices, values=values, dense_shape=[batch_size, 41])
+    )
+    out = reco_model(input_tensor, labels=input_labels, training=True)
+    assert isinstance(out, tf.Tensor)
+    assert isinstance(reco_model, tf.keras.Model)
+    assert out.numpy().shape == (batch_size, *output_size)
+
+
```

Comment:
We'll have to refactor this to test the training call of all recognition models as well. But we can keep this for another PR!

Mind adding a few comments about the input_labels generation?